File: 237/237.py
# Definition for singly-linked list.

# class ListNode(object):

#     def __init__(self, x):

#         self.val = x

#         self.next = None



class Solution(object):

    def deleteNode(self, node):
        """
        :type node: ListNode
        :rtype: void Do not return anything, modify node in-place instead.
        """
        if node == None or node.next == None:
            return

        cur = node
        pre = None

        while(cur.next != None):
            cur.val = cur.next.val
            pre = cur
            cur = cur.next

        pre.next = None
        del cur

File: 237/237.cpp
/*
   Write a function to delete a node (except the tail) in a singly linked list, given only access to that node.

   Supposed the linked list is 1 -> 2 -> 3 -> 4 and you are given the third node with value 3, the linked list should become 1 -> 2 -> 4 after calling your function. 

*/


/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */

class Solution {

    public:
        void deleteNode(ListNode* node) 
        {
            if (node == nullptr || node->next == nullptr) return; 

            ListNode *cur, *pre;
            pre = nullptr;
            cur = node;

            while(cur->next)
            {
                cur->val = cur->next->val;
                pre = cur;
                cur = cur->next;
            }

            pre->next = nullptr;
            free(cur);
        }

};
